pkg/scheduler/core/scheduling_queue.go

```diff
@@ -169,12 +182,20 @@ type PriorityQueue struct {
 // Making sure that PriorityQueue implements SchedulingQueue.
 var _ = SchedulingQueue(&PriorityQueue{})
 
-func NewPriorityQueue() *PriorityQueue {
+func NewPriorityQueue(stopEverything <-chan struct{}) *PriorityQueue {
+	return NewPriorityQueueWithClock(stopEverything, util.RealClock{})
+}
+
+func NewPriorityQueueWithClock(stopEverything <-chan struct{}, clock util.Clock) *PriorityQueue {
 	pq := &PriorityQueue{
-		activeQ:        newHeap(cache.MetaNamespaceKeyFunc, util.HigherPriorityPod),
+		stop:           stopEverything,
+		clock:          clock,
+		activeQ:        util.NewHeap(cache.MetaNamespaceKeyFunc, util.HigherPriorityPod),
 		unschedulableQ: newUnschedulablePodsMap(),
+		podBackoff:     util.CreateDefaultPodBackoffWithClock(clock),
```

Comment:
I think we should set the max duration to 15 seconds. 60 seconds is too long and given the exponential growth of backoff time, it does not take that many tries to get to 60 seconds.